document-store/src/main/java/org/hypertrace/core/documentstore/mongo/parser/MongoSelectingExpressionParser.java

```diff
@@ -79,4 +83,12 @@ public static BasicDBObject getProjectClause(final Query query) {
 
     return spec.getExpression().visit(parser);
   }
+
+  private static <T> T mergeValues(final T first, final T second) {
+    if (first.equals(second)) {
+      return second;
+    }
+
+    throw new IllegalArgumentException("Query contains duplicate aliases");
```

Comment:
didn't understand the exception message. Won't duplicate selections have duplicate aliases?

Reply:
There can be 3 cases as follows.
1. Same selection and same alias as in `SELECT sum(price) AS total, sum(price) AS total`. In this case, both the keys (alias) and both the values (selection) match. Key: total, Value: sum(price). In this case, we pick the second value (equivalently we could have picked the first)
2. Same selection with different alias as in `SELECT sum(price) AS total, sum(price) AS price`. In this case, we don't have a problem since the keys (aliases) are unique and we wouldn't be hitting this method
3. Different selections with same aliases as in `SELECT SUM(quantity) AS total, SUM(price) AS total`. We shouldn't be allowing such cases as it clearly invites ambiguity. When the parsed values are different for the same key (alias), we detect it and throw exception.

Reply:
Updating the message for clarity.